### PR TITLE
fixes dual wielding rifles and whatnot

### PR DESCRIPTION
**Pull request by FrownTown**
## About The Pull Request
<details>
<summary>
	Fixes Dual Wielding 
</summary>
<hr>

Makes it impossible to dual wield (fire simultaneously using harm intent) weapons that should logically be fired while carried with two hands, including most rifles, shotguns, and two-handed SMGs (you can tell the difference between an SMG and a machinepistol by their sprite, the molly can be dual-wielded, but not the straylight).
	
<hr>
</details>

## Changelog
:cl:

/:cl:

### CODE REVIEW

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -200,13 +202,15 @@
...
+		else if(off_hand && off_hand.can_hit(user))
 			spawn(1)
 			off_hand.Fire(A,user,params)
-	else
-		dual_wielding = FALSE
+		else
+			dual_wielding = FALSE
 
-	Fire(A,user,params) //Otherwise, fire normally.
+		Fire(A,user,params) //Otherwise, fire normally.
```
Comment (PlinHost) [inline code/modules/projectiles/gun.dm]: This should not be indented. It makes guns impossible to fire on any intent but harm.

File: code/modules/projectiles/gun.dm [hunk 2]
```diff
@@ -200,13 +202,15 @@
...
-		if(off_hand && off_hand.can_hit(user))
+		if(!can_dual)
+			dual_wielding = FALSE
+		else if(off_hand && off_hand.can_hit(user))
 			spawn(1)
 			off_hand.Fire(A,user,params)
-	else
-		dual_wielding = FALSE
+		else
+			dual_wielding = FALSE
```
Comment (PlinHost) [inline code/modules/projectiles/gun.dm]: This too. It could lead to `dual_wielding` not being reset.

File: code/modules/projectiles/gun.dm (hunk 1 above)
Comment (FrownTown) [reply code/modules/projectiles/gun.dm]: fixed!

File: code/modules/projectiles/gun.dm (hunk 2 above)
Comment (FrownTown) [reply code/modules/projectiles/gun.dm]: fixed this too! 

Thanks for the help, still new, still learning, appreciate all the advice I get

**Comment by TorinoFermic** [comment]
You'll need to push your new commit to this branch

**Comment by FrownTown** [comment]
> You'll need to push your new commit to this branch

Yeah sorry about that, thought I hit the button before I got up to grab breakfast but I must have missed it lol

**Pull request merged**